src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java

```diff
@@ -0,0 +1,348 @@
+	public TypeAttribute getTypeAttribute(long id) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<TypeAttribute> query = session.createQuery("FROM TypeAttributeImpl typeAttribute WHERE typeAttribute.id = :id",
+				TypeAttribute.class)
+				.setParameter("id", id);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, null);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<TypeAttribute> getTypeAttributes(long typeId) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<TypeAttribute> query = session.createQuery("FROM TypeAttributeImpl typeAttribute WHERE typeAttribute.typeId = :typeId",
+				TypeAttribute.class)
+				.setParameter("typeId", typeId);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Entity getEntity(long id) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Entity> query = session.createQuery("FROM EntityImpl entity WHERE entity.id = :id",
+				Entity.class)
+				.setParameter("id", id);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, null);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Entity> getEntities(long parentId, long entityTypeId) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Entity> query = session.createQuery("FROM EntityImpl entity "
+				+ "WHERE entity.typeId = :entityTypeId AND entity.parentId = :parentId",
+				Entity.class)
+				.setParameter("entityTypeId", entityTypeId)
+				.setParameter("parentId", parentId);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<EntityType> getEntityTypes() {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<EntityType> query = session.createQuery("FROM EntityTypeImpl entityType",
+				EntityType.class);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Event> getEvents(long entityOwnerId, long eventTypeId, int listStartIndex, int listEndIndex) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Event> query = session.createQuery("FROM EventImpl event "
+				+ "WHERE event.typeId = :eventTypeId AND event.entityId = :entityOwnerId",
+				Event.class)
+				.setParameter("eventTypeId", eventTypeId)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "eventTypeId" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)